index.js

```diff
@@ -12,8 +17,15 @@ function init() {
 
 	const selectedText = editor.getSelectedText();
 	const text = selectedText || editor.getText();
+	const config = atom.config.get('perfectionist');
+
+	config.syntax = 'scss';
 
-	postcss(perfectionist(atom.config.get('perfectionist'))).process(text, {
+	if (editor.getGrammar().scopeName === 'source.css') {
+		delete config.syntax;
+	}
```

Comment:
I was doing that before, essentially, and you said it should default to scss all the time to work with unsaved buffers. Not sure how this is any different than what I had previously.